src/bot/commands/lastfm/fm.ts

```diff
@@ -207,19 +214,47 @@ export class LastFm extends Command {
                                             value: artist.stats.playcount.replace(/\B(?=(\d{3})+(?!\d))/g, ",")
                                         }
                                     ]
-                                }                            
+                                }
                             })
                         })
 
                     });
+                } else {
+                    input.channel.send('Please input an artist name. `lastfm artist "<Artist Name>"`');
                 }
                 break;
             case 'chart':
+            case 'albumchart':
                 if (user && user != '') {
-                    let time = new Date().getTime()
-                    let requestURL = request((lastfmURL + 'user.getWeeklyAlbumChart' + '&user=' + user + '&from=946688400&to=1552697146' + '&api_key= ' + key + '&limit=2&format=json'), async (error: any, response: Response, body: any) => {
+                    let currentTime = (Math.floor(new Date().getTime()/1000.0));
+                    let from = currentTime-315569260;
+                    if (input.getArgument('user')) {
+                        switch(user) {
+                            case 'week':
+                            case 'weekly':
+                                from = currentTime-604800;
+                                break;
+                            case 'month':
+                            case 'monthly':
+                                from = currentTime-2629743;
+                                break;
+                            case 'year':
+                            case 'yearly':
+                                from = currentTime-31556926;
+                                break;
+                            case 'all':
```

Comment:
Manually setting the time for 'all', 'alltime', and 'default' seems redundant here, since it just sets it to the default value for "from". Correct me if I'm wrong, but I don't think removing lines 247, 248, or 250 would change the way the code functions.